examples/How_to_use_moderation.ipynb

```diff
@@ -0,0 +1,455 @@
+    ", a tool you can use to check whether text is potentially harmful.\n",
+    "\n",
+    "This notebook will concentrate on:\n",
+    "\n",
+    "- **Input Moderation:** Identifying and flagging inappropriate or harmful content before it is processed by your LLM.\n",
+    "- **Output Moderation:** Reviewing and validating the content generated by your LLM before it reaches the end user.\n",
+    "- **Custom Moderation:** Tailoring moderation criteria and rules to suit the specific needs and context of your application, ensuring a personalized and effective content control mechanism."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from openai import OpenAI\n",
+    "client = OpenAI()\n",
+    "\n",
+    "GPT_MODEL = 'gpt-3.5-turbo'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 1. Input moderation\n",
+    "Input Moderation focuses on preventing harmful or inappropriate content from reaching the LLM, with common applications including:\n",
+    "- **Content Filtering:** Prevent the spread of harmful content such as hate speech, harassment, explicit material, and misinformation on social media, forums, and content creation platforms.\n",
+    "- **Community Standards Enforcement:** Ensure that user interactions, such as comments, forum posts, and chat messages, adhere to the community guidelines and standards of online platforms, including educational environments, gaming communities, or dating apps.\n",
+    "- **Spam and Fraud Prevention:** Filter out spam, fraudulent content, and misleading information in online forums, comment sections, e-commerce platforms, and customer reviews.\n",
+    "\n",
+    "These measures act as preventive controls, operating before or alongside the LLM to alter your application's behavior if specific criteria are met."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Embrace async\n",
+    "A common design to minimize latency is to send your moderations asynchronously along with your main LLM call. If your moderation gets triggered you send back a placeholder response, otherwise send back the LLM response. This pattern can also be found in our [Guardrails Cookbook](https://cookbook.openai.com/examples/how_to_use_guardrails).\n",
+    "\n",
+    "We'll use this approach, creating an execute_chat_with_moderation function that will run our LLM's get_chat_response and the check_expression moderation function in parallel, and return the LLM response only if the moderation returns False (not triggered).\n",
+    "\n",
+    "#### Workflow:\n",
+    "We will create a workflow that incorporates the Moderation API to check user input for any potentially unsafe content before it is sent to the Language Model (LLM). This ensures that only appropriate content is processed further in the application.\n",
```

Comment:
Oh interesting... it used to be that we only allowed the use of our moderation API for text generated by our models, but I don't actually see that restriction in the docs anymore. Might still be good to double check this with product!

Reply:
they confirmed we support it!